packages/reactive-element/src/reactive-element.ts

```diff
@@ -817,6 +816,7 @@ export abstract class ReactiveElement
       this._initializers = [...superCtor._initializers];
     }
     this.elementProperties = new Map(superCtor.elementProperties);
+    this.__collectMetadata();
```

Comment:
I'm curious how this isn't breaking tests. Are we lacking some coverage? `finalize()` can be called from `addInitializer()` and `createProperty()`. If either of those are called before `get observedAttributes` and before all the decorators are processed, then we should only convert a subset of the decorator metadata to elementProperties.

I think we at least need a test that has a decorator application that adds an initializer followed by `@property()` decorators.

Reply:
Completely agree with you. In the material-web case, the decorator on their super class is actually a `@customElement` decorator, which then results in every single property losing reactivity (since properties are only set in super classes).

Will add more tests.